x-pack/plugins/cloud_security_posture/public/pages/vulnerabilities/vulnerabilities_by_resource/vulnerabilities_by_resource.tsx

```diff
@@ -0,0 +1,345 @@
+import { ErrorCallout } from '../../configurations/layout/error_callout';
+import { FindingsSearchBar } from '../../configurations/layout/findings_search_bar';
+import { useLimitProperties } from '../../../common/utils/get_limit_properties';
+import { LimitedResultsBar } from '../../configurations/layout/findings_layout';
+import {
+  getVulnerabilitiesByResourceColumnsGrid,
+  vulnerabilitiesByResourceColumns,
+} from './vulnerabilities_by_resource_table_columns';
+import {
+  defaultLoadingRenderer,
+  defaultNoDataRenderer,
+} from '../../../components/cloud_posture_page';
+import { getFilters } from '../utils/get_filters';
+import { FILTER_IN, FILTER_OUT, SEARCH_BAR_PLACEHOLDER, VULNERABILITIES } from '../translations';
+import { useStyles } from '../hooks/use_styles';
+import { FindingsGroupBySelector } from '../../configurations/layout/findings_group_by_selector';
+import { vulnerabilitiesPathnameHandler } from '../utils/vulnerabilities_pathname_handler';
+import { useLatestVulnerabilitiesByResource } from '../hooks/use_latest_vulnerabilities_by_resource';
+import { EmptyState } from '../../../components/empty_state';
+import { SeverityMap } from './severity_map';
+import { VULNERABILITY_RESOURCE_COUNT } from './test_subjects';
+
+const getDefaultQuery = ({ query, filters }: any): any => ({
+  query,
+  filters,
+  sort: [{ id: vulnerabilitiesByResourceColumns.vulnerabilities_count, direction: 'desc' }],
+  pageIndex: 0,
+});
+
+export const VulnerabilitiesByResource = ({ dataView }: { dataView: DataView }) => {
+  const {
+    pageIndex,
+    query,
+    sort,
+    queryError,
+    pageSize,
+    onChangeItemsPerPage,
+    onChangePage,
+    onSort,
+    urlQuery,
+    setUrlQuery,
+    onResetFilters,
+  } = useCloudPostureTable({
+    dataView,
+    defaultQuery: getDefaultQuery,
+    paginationLocalStorageKey: LOCAL_STORAGE_PAGE_SIZE_FINDINGS_KEY,
+  });
+
+  const styles = useStyles();
+  const { data, isLoading, isFetching } = useLatestVulnerabilitiesByResource({
+    query,
+    sortOrder: sort[0]?.direction,
+    enabled: !queryError,
+    pageIndex,
+    pageSize,
+  });
+
+  const { isLastLimitedPage, limitedTotalItemCount } = useLimitProperties({
+    total: data?.total,
+    pageIndex,
+    pageSize,
+  });
+
+  const columns = useMemo(() => {
+    const getColumnIdValue = (rowIndex: number, columnId: string) => {
+      const vulnerabilityRow = data?.page[rowIndex];
+      if (!vulnerabilityRow) return null;
+
+      if (columnId === vulnerabilitiesByResourceColumns.resource_id) {
+        return vulnerabilityRow['resource.id'];
+      }
+      if (columnId === vulnerabilitiesByResourceColumns.resource_name) {
+        return vulnerabilityRow['resource.name'];
+      }
+      if (columnId === vulnerabilitiesByResourceColumns.region) {
+        return vulnerabilityRow['cloud.region'];
+      }
+    };
+
+    const cellActions: EuiDataGridColumnCellAction[] = [
```

Comment:
might be better to export from its own file

Reply:
There are some common functions to be reused with the default vulnerability table as well, I think it can be a refactor candidate for later